diff --git a/RenderSystems/GL3Plus/src/OgreGL3PlusTextureGpu.cpp b/RenderSystems/GL3Plus/src/OgreGL3PlusTextureGpu.cpp
index c9a1147b188..7e027a676bd 100644
--- a/RenderSystems/GL3Plus/src/OgreGL3PlusTextureGpu.cpp
+++ b/RenderSystems/GL3Plus/src/OgreGL3PlusTextureGpu.cpp
@@ -619,6 +619,8 @@ namespace Ogre
         }
     }
     //-----------------------------------------------------------------------------------
+    GLAPI void APIENTRY glCopyImageSubDataNV (GLuint srcName, GLenum srcTarget, GLint srcLevel, GLint srcX, GLint srcY, GLint srcZ, GLuint dstName, GLenum dstTarget, GLint dstLevel, GLint dstX, GLint dstY, GLint dstZ, GLsizei width, GLsizei height, GLsizei depth);
+    //-----------------------------------------------------------------------------------
     void GL3PlusTextureGpu::copyTo( TextureGpu *dst, const TextureBox &dstBox, uint8 dstMipLevel,
                                     const TextureBox &srcBox, uint8 srcMipLevel,
                                     bool keepResolvedTexSynced,
@@ -649,15 +651,16 @@ namespace Ogre
                                           dstBox.getZOrSlice() + dstGl->getInternalSliceStart(),
                                           srcBox.width, srcBox.height, srcBox.getDepthOrSlices() ) );
             }
-            /*TODO
             else if( support.checkExtension( "GL_NV_copy_image" ) )
             {
                 OCGE( glCopyImageSubDataNV( this->mFinalTextureName, this->mGlTextureTarget,
-                                            srcMipLevel, srcBox.x, srcBox.y, srcBox.z,
-                                            dstGl->mFinalTextureName, dstGl->mGlTextureTarget,
-                                            dstMipLevel, dstBox.x, dstBox.y, dstBox.z,
-                                            srcBox.width, srcBox.height, srcBox.getDepthOrSlices() ) );
-            }*/
+                                          srcMipLevel, srcBox.x, srcBox.y,
+                                          srcBox.getZOrSlice() + this->getInternalSliceStart(),
+                                          dstGl->mFinalTextureName, dstGl->mGlTextureTarget,
+                                          dstMipLevel, dstBox.x, dstBox.y,
+                                          dstBox.getZOrSlice() + dstGl->getInternalSliceStart(),
+                                          srcBox.width, srcBox.height, srcBox.getDepthOrSlices() ) );
+            }
             /*TODO: These are for OpenGL ES 3.0+
             else if( support.checkExtension( "GL_OES_copy_image" ) )
             {
